pkg/build/controller/build/build_controller.go

```diff
@@ -2333,6 +2333,46 @@ func isValidTransition(from, to buildv1.BuildPhase) bool {
 	return true
 }
 
+func checkLogSnippetsForBadBuildPhase(build *buildv1.Build, update *buildUpdate) bool {
+	switch build.Status.Phase {
+	case buildv1.BuildPhaseFailed:
+		return true
+	case buildv1.BuildPhaseError:
+		// containers completing but exiting with non-0 RCs fall under here
+		return true
+	}
+	if update != nil && update.phase != nil {
+		switch *update.phase {
+		case buildv1.BuildPhaseFailed:
+			return true
+		case buildv1.BuildPhaseError:
+			return true
+		}
+	}
+	return false
+}
+
+func checkLogSnippetForBadPodStatus(pod *corev1.Pod) (bool, bool, int) {
+	if pod == nil {
+		return false, false, -1
+	}
+	if len(pod.Status.ContainerStatuses) != 0 && pod.Status.ContainerStatuses[0].State.Terminated != nil {
+		return true, false, -1
+	}
+	// we have 1 container, but up to 3 init containers, so for readability factoring out the init container
+	// terminated determination here
+	initContainerTerminated := -1
+	if len(pod.Status.InitContainerStatuses) > 0 {
+		for i, ics := range pod.Status.InitContainerStatuses {
+			if ics.State.Terminated != nil {
+				initContainerTerminated = i
+				return false, true, initContainerTerminated
+			}
+		}
+	}
+	return false, false, initContainerTerminated
```

Comment:
```suggestion
	return false, false, -1
```
it logically can't be any other value

Reply:
part of next push